kv/util/util.go

```diff
@@ -33,54 +34,179 @@ import (
 var errNilStore = errors.New("kv store is nil")
 
 type getValueFn func(kv.Value) (interface{}, error)
+
 type updateFn func(interface{})
 
+type validateBoundsFn func(interface{}) (lower, upper interface{}, err error)
```

Comment:
Yea, I'll update this. I went back and forth myself. Think a more agnostic `validate` is the way to go.